lib/cylc/job_file.py

```diff
@@ -31,6 +31,9 @@ class JobFileWriter(object):
 
     """Write task job files."""
 
+    env_match1 = re.compile(r"^(~[^/\s]*/)(.*)$")
+    env_match2 = re.compile(r"^~[^\s]*$")
```

Comment:
We capitalize constants like this, by convention - however, we don't need them at all in this case (see subsequent comments...)